overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -238,6 +245,114 @@ func runServiceCommand(context *hookstate.Context, inst *servicestate.Instructio
 	}
 }
 
+func validateSnapAndCompsNames(names []string, ctxSnap string) ([]string, error) {
+	var allComps []string
+	for _, name := range names {
+		snap, comps := snap.SplitSnapInstanceAndComponents(name)
+		// if snap is present it must be the context snap for the moment
+		if snap != "" && snap != ctxSnap {
+			return nil, errors.New("cannot install snaps using snapctl")
+		}
+		for _, comp := range comps {
+			if err := naming.ValidateSnap(comp); err != nil {
+				return nil, err
+			}
+		}
+		allComps = append(allComps, comps...)
+	}
+	return allComps, nil
+}
+
+type managementCommandTp int
```

Comment:
wouldn't op/operation be more appropriate than tp/type ?

Reply:
Changed